src/js/utils/ui.js

```diff
@@ -0,0 +1,128 @@
+            elem.addEventListener('touchmove', interactDragHandler);
+            elem.addEventListener('touchcancel', interactEndHandler);
+            elem.addEventListener('touchend', interactEndHandler);
+
+            document.addEventListener('mousemove', interactDragHandler);
+            document.addEventListener('mouseup', interactEndHandler);
+        }
+
+        function interactDragHandler(evt) {
+            var touchEvents = events.touchEvents;
+
+            if (_hasMoved) {
+                triggerEvent(touchEvents.DRAG, evt);
+            } else {
+                triggerEvent(touchEvents.DRAG_START, evt);
+                _hasMoved = true;
+                triggerEvent(touchEvents.DRAG, evt);
+            }
+        }
+
+        function interactEndHandler(evt) {
+            var touchEvents = events.touchEvents;
+
+            elem.removeEventListener('touchmove', interactDragHandler);
+            elem.removeEventListener('touchcancel', interactEndHandler);
+            elem.removeEventListener('touchend', interactEndHandler);
+
+            document.removeEventListener('mousemove', interactDragHandler);
+            document.removeEventListener('mouseup', interactEndHandler);
+
+            if (_hasMoved) {
+                triggerEvent(touchEvents.DRAG_END, evt);
+            } else {
+                // This allows the controlbar/dock/logo click events not to be forwarded to the view
+                evt.cancelBubble = true;
+                if(evt instanceof MouseEvent) {
+                    triggerEvent(touchEvents.CLICK, evt);
+                } else {
+                    triggerEvent(touchEvents.TAP, evt);
+                }
+            }
+
+            _hasMoved = false;
+        }
+
+        function normalizeUIEvent(type, srcEvent) {
+            var source;
+            if(srcEvent instanceof MouseEvent) {
+                source = srcEvent;
+            } else {
+                if (srcEvent.touches && srcEvent.touches.length) {
+                    source = srcEvent.touches[0];
+                } else {
+                    source = srcEvent.changedTouches[0];
+                }
+            }
+            return {
+                type: type,
+                target: srcEvent.target,
+                currentTarget: _elem,
+                pageX: source.pageX,
+                pageY: source.pageY
+            };
+        }
+
+        // Preventdefault to prevent click events
+        function preventDefault(evt) {
+            if (evt.preventManipulation) {
+                evt.preventManipulation();
+            }
+            if (evt.preventDefault) {
+                evt.preventDefault();
+            }
+        }
+
+        var self = this;
+        function triggerEvent(type, srcEvent) {
+            preventDefault(srcEvent);
+            if( _enableDoubleTap && (type === events.touchEvents.CLICK || type === events.touchEvents.TAP)){
+                if(Date.now() - _lastClickTime < _doubleClickDelay) {
```

Comment:
Date.now() doesn't work in IE8 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now

_.now() is a good abstraction for this